src/main/java/pl/sunflux/parking/ParkingStopImpl.java

```diff
@@ -0,0 +1,70 @@
+package pl.sunflux.parking;
+
+import org.joda.time.DateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import pl.sunflux.controllers.rest.containers.driver.StopParkingContainer;
+import pl.sunflux.controllers.rest.responses.driver.StopParkingResponse;
+import pl.sunflux.entity.*;
+import pl.sunflux.repository.*;
+
+@Service
+public class ParkingStopImpl implements ParkingStopInterface {
+
+    private final ParkingMeterUsageRepository parkingMeterUsageRepository;
+    private final DriverRepository driverRepository;
+    private final VehicleRepository vehicleRepository;
+    private final ParkingMeterRepository parkingMeterRepository;
+    private final ParkingFeeRepository parkingFeeRepository;
+    private final ParkingRatesCalculationImpl parkingRatesCalculation;
+
+    @Autowired
+    public ParkingStopImpl(ParkingMeterUsageRepository parkingMeterUsageRepository, DriverRepository driverRepository, VehicleRepository vehicleRepository, ParkingMeterRepository parkingMeterRepository, ParkingFeeRepository parkingFeeRepository, ParkingRatesCalculationImpl parkingRatesCalculation) {
+        this.parkingMeterUsageRepository = parkingMeterUsageRepository;
+        this.driverRepository = driverRepository;
+        this.vehicleRepository = vehicleRepository;
+        this.parkingMeterRepository = parkingMeterRepository;
+        this.parkingFeeRepository = parkingFeeRepository;
+        this.parkingRatesCalculation = parkingRatesCalculation;
+    }
+
+    @Override
+    public StopParkingResponse prepareResponse(StopParkingContainer stopParkingContainer) {
+        Driver driver = driverRepository.findByDriverIdCard(stopParkingContainer.getDriverIdCard());
+        Vehicle vehicle = vehicleRepository.findBySerialNumberAndDriver(stopParkingContainer.getVehicleSerialNumber(), driver);
+        ParkingMeterUsage parkingMeterUsage = parkingMeterUsageRepository.findByIdAndVehicleAndDateEndIsNull(stopParkingContainer.getParkingTicketId(), vehicle);
+
+        StopParkingResponse stopParkingResponse = new StopParkingResponse();
+
+        if (parkingMeterUsage == null) {
+            return stopParkingResponse;
+        }
+
+        DateTime dateEnd = DateTime.now();
+        parkingMeterUsage.setDateEnd(dateEnd.toDate());
+        parkingMeterUsageRepository.save(parkingMeterUsage);
+
+        saveParkingFee(parkingMeterUsage);
+
+        ParkingMeter parkingMeter = parkingMeterUsage.getParkingMeter();
+        parkingMeter.setFree(true);
```

Comment:
Czemu przechowujesz wolne liczniki parkingowe? Jaką to daje przewagę w stosunku do usuwania zwolnionych?